vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -1,207 +1,259 @@
+            return space.get();
+        }
+        return null;
+    }
+
+    @Override
+    public ISpace getFullyLoadedSpace(String id) {
+        ISpace space = getSpace(id);
+        if (space == null) {
+            return null;
+        }
+        // load lazy collections
+        space.getSpaceLinks().size();
+        space.getModuleLinks().size();
+        space.getExternalLinks().size();
+        return space;
+    }
+
+    @Override
+    public List<ISpace> getAllSpaces() {
+        List<ISpace> spaces = new ArrayList<>();
+        spaceRepo.findAll().forEach(s -> spaces.add(s));
+        return spaces;
+    }
+    
+    
+    @Override
+    public List<ISpace> getSpacesWithStatus(SpaceStatus status) {
+        List<ISpace> spaces = new ArrayList<>();
+        spaceRepo.findAllBySpaceStatus(status).forEach(s -> spaces.add(s));
+        return spaces;
+    }
+
+	/**
+	 * Method to delete space based on id
+	 * 
+	 * @param id if id is null throws exception, else delete corresponding space
+	 * @throws SpaceDoesNotExistException
+	 */
+    @Override
+    public void deleteSpaceById(String id) throws SpaceDoesNotExistException {
+        try {
+            // Order of deletion: Tables: SPACELINKDISPLAY (link_id), SPACELINK
+            // (source_space_id), SPACEDISPLAY (space_id), SPACE(id)
+
+            List<SpaceLink> spaceLinks = spaceLinkRepo.getLinkedSpaces(id);
+            List<SpaceLink> fromSpaceLinks = spaceLinkRepo.getLinkedFromSpaces(id);
+            Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition();
+            // When space has other links attached to it
+            if (spaceLinks.size() > 0) {
+                // To delete links that access to the space getting deleted and replacing it as null
+                for(SpaceLink sp : fromSpaceLinks) {
+                    sp.setTargetSpace(null);
+                    spaceLinkRepo.save(sp);
+                }
+                // To delete the links on the space getting deleted
+                for(SpaceLink sp : spaceLinks) {
+                    spaceLinkDisplayRepo.deleteBySpaceLinkId(sp.getId());
+                }
+                spaceLinkRepo.deleteBySourceSpaceId(id);
+            }
+            // If the space is startSpace, we delete the space from the exhibition first.
+            if(exhibition != null && exhibition.getStartSpace() != null
+                    && exhibition.getStartSpace().getId().equalsIgnoreCase(id)) {
+                exhibition.setStartSpace(null);
+                exhibitRepo.save(exhibition);
+            }
+            // When space has no other links attached to it
+            spaceDisplayRepo.deleteBySpaceId(id);
+            spaceRepo.deleteById(id);
+
+        } catch (IllegalArgumentException exception) {
+            logger.error("Sorry, some problem occurred while deleting space.", exception);
+        }
+    }
+
+    @Override
+    public List<SpaceLink> isLinkedFromSpace(String id) {

```

Comment:
the method name indicates it returns a boolean, which it doesn't. needs a better method name.